loopin.go

```diff
@@ -394,20 +397,35 @@ func (s *loopInSwap) executeSwap(globalCtx context.Context) error {
 func (s *loopInSwap) waitForHtlcConf(globalCtx context.Context) (
 	*chainntnfs.TxConfirmation, error) {
 
+	// Register for confirmation of the htlc. It is essential to specify not
+	// just the pk script, because an attacker may publish the same htlc
+	// with a lower value and we don't want to follow through with that tx.
+	// In the unlikely event that our call to SendOutputs crashes and we
+	// restart, htlcTxHash will be nil at this point. Then only register
```

Comment:
Not for this PR, and depending on how quickly this gets addresses in lnd, but would it be possible for us to try to recover our txid from lnd in the case of this crash? Since we know our pksrcipt, we can `listChainTxns` and check out the outputs on the raw tx?

Reply:
I believe the real fix here is to separate construction of the tx and publication. With the new utxo locking API in 0.11 that will be possible. Then we'll always first persist the tx hash and then publish and this gap will disappear.